.github/workflows/bench.yml

```diff
@@ -1,52 +1,135 @@
+            echo "Setup completed for ${{ matrix.service }}"
+            else
+              echo "No specific setup script found for ${{ matrix.service }}, running general setup"
+            fi
+
+            echo "Setup completed"
+            echo "Running benchmark for ${{ matrix.service }}"
+            chmod +x ./run_benchmarks.sh
+            valid_services=("apollo_server" "caliban" "netflix_dgs" "gqlgen" "tailcall" "async_graphql" "hasura" "graphql_jit")
+            if [[ ! " ${valid_services[@]} " =~ " ${{ matrix.service }} " ]]; then
+                echo "Error: Invalid service name '${{ matrix.service }}'"
+                exit 1
+            fi
+            if ! ./run_benchmarks.sh ${{ matrix.service }}; then
+                echo "Error: Benchmark failed for ${{ matrix.service }}"
+                exit 1
+            fi
+            echo "Benchmark completed for ${{ matrix.service }}"
+
+
+      - name: List benchmark files
+        run: |
+          echo "Contents of current directory:"
+          ls -la
+          echo "Matching bench*.txt files:"
+          ls -la bench*.txt || echo "No matching files found"
+
+          
+      - name: Upload benchmark results
+        uses: actions/upload-artifact@v3
+        with:
+          name: benchmark-results
+          path: bench*.txt
+
+
+  analyze:
+    needs: benchmark
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout (GitHub)
+        uses: actions/checkout@v4
+      
+      - name: Download all benchmark results
+        uses: actions/download-artifact@v3
+        with:
+          name: benchmark-results
+          path: .
+      
+      - name: List downloaded artifacts
+        run: ls -la bench*.txt || echo "No matching files found"
+      
+      - name: output of tailcall
+        run: |
+          echo "Benchmark 1"
+          cat ./bench1_result1_graphql_tailcall_run.sh.txt
+          cat ./bench1_result2_graphql_tailcall_run.sh.txt
+          cat ./bench1_result3_graphql_tailcall_run.sh.txt
+          echo "End of Becnhmark 1"
+          echo ""
+          echo "Benchmark 2"
+          cat ./bench2_result1_graphql_tailcall_run.sh.txt
+          cat ./bench2_result2_graphql_tailcall_run.sh.txt
+          cat ./bench2_result3_graphql_tailcall_run.sh.txt
+          echo "End of Becnhmark 2"
+          echo ""
+          echo "Becnhmark 3"
+          cat ./bench3_result1_graphql_tailcall_run.sh.txt
+          cat ./bench3_result2_graphql_tailcall_run.sh.txt
+          cat ./bench3_result3_graphql_tailcall_run.sh.txt
+          echo "End of Benchmark 2"
+          echo ""
+
+      - name: Analyze results
+        run: |
+          rm results.md
+          for bench in 1 2 3; do
+            ls bench${bench}*.txt || echo "No matching files found for bench${bench}*.txt"
+            echo "Processing files for bench${bench}:"
+            bash analyze.sh bench${bench}*.txt
+            echo bench${bench}*.txt
+          done
```

Comment:
Move all this logic into a shell script.

Reply:
oh okay